File: arrayDiff.js
function arrayDiff(a, b) {
    for (let element of b) {
        let index = 0;
        while (index < a.length) a[index] === element ? a.splice(index, 1) : ++index
    }
    return a
}

function array_diff(a, b) {
    return a.filter(e => !b.includes(e));
  }

console.log(array_diff([1, 2, 2, 2, 3], [2]))

File: array_diff.py
"""https://www.codewars.com/kata/523f5d21c841566fde000009"""

def array_diff(a, b):
    """
    Your goal in this kata is to implement a difference function, 
    which subtracts one list from another and returns the result.

    Args:
        a (list): list with integers
        b (list): list with integers

    Returns:
        list: list with integers
    """
    return list(filter( lambda x: x not in b , a))

print(array_diff([1, 2, 3], [2]))
